### PR TITLE
finalize adding eltypes to show

### PR DESCRIPTION
**Pull request by bkamins**
This is a follow-up to #2142 completing the changes.

(as this is a clean-up PR I have also added some minor changes to unrelated code - specifically I have changed comments giving an expected return types of functions to assertions)

### CODE REVIEW

File: src/abstractdataframe/show.jl [hunk 1]
```diff
@@ -155,7 +160,7 @@ function getmaxwidths(df::AbstractDataFrame,
                       rowindices2::AbstractVector{Int},
                       rowlabel::Symbol,
                       rowid=nothing,
-                      show_eltype::Bool=true) # -> Vector{Int}
+                      show_eltype::Bool=true)::Vector{Int}
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]:
```suggestion
                      show_eltype::Bool=true)
```

File: src/abstractdataframe/show.jl [hunk 2]
```diff
@@ -222,7 +227,7 @@ julia> DataFrames.getprintedwidth(maxwidths)
 15
 ```
 """
-function getprintedwidth(maxwidths::Vector{Int}) # -> Int
+function getprintedwidth(maxwidths::Vector{Int})::Int
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]:
```suggestion
function getprintedwidth(maxwidths::Vector{Int})
```

File: src/abstractdataframe/show.jl [hunk 3]
```diff
@@ -275,7 +280,7 @@ julia> DataFrames.getchunkbounds(maxwidths, true, displaysize()[2])
 """
 function getchunkbounds(maxwidths::Vector{Int},
                         splitcols::Bool,
-                        availablewidth::Int) # -> Vector{Int}
+                        availablewidth::Int)::Vector{Int}
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]:
```suggestion
                        availablewidth::Int)
```

File: src/abstractdataframe/show.jl [hunk 4]
```diff
@@ -340,7 +345,7 @@ function showrowindices(io::IO,
                         maxwidths::Vector{Int},
                         leftcol::Int,
                         rightcol::Int,
-                        rowid) # -> Void
+                        rowid)::Nothing
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]:
```suggestion
                        rowid)
```

File: src/abstractdataframe/show.jl [hunk 5]
```diff
@@ -458,7 +463,7 @@ function showrows(io::IO,
                   rowlabel::Symbol = :Row,
                   displaysummary::Bool = true,
                   eltypes::Bool = true,
-                  rowid=nothing) # -> Void
+                  rowid=nothing)::Nothing
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]:
```suggestion
                  rowid=nothing)
```

**Comment by bkamins** [comment]
Good point - indeed we will have correct type inference in all these cases. I applied your suggestions.

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**